contracts/StakingContract.sol

```diff
@@ -296,28 +297,31 @@ contract StakingContract is IStakingContract {
     }
 
     /// @dev Migrates the stake of msg.sender from this staking contract to a new approved staking contract.
-    function migrateStakedTokens(IStakingContract _newStakingContract) external onlyWhenStakesNotReleased {
+    /// @param _amount uint256 The amount of tokens to migrate.
+    function migrateStakedTokens(IStakingContract _newStakingContract, uint256 _amount) external onlyWhenStakesNotReleased {
         require(isApprovedStakingContract(_newStakingContract),
             "StakingContract::migrateStakedTokens - migration destination wasn't approved");
+        require(_amount > 0, "StakingContract::migrateStakedTokens - amount must be greater than 0");
 
         address stakeOwner = msg.sender;
         Stake storage stakeData = stakes[stakeOwner];
-        uint256 amount = stakeData.amount;
+        uint256 stakedAmount = stakeData.amount;
 
-        require(amount > 0, "StakingContract::migrateStakedTokens - no staked tokens");
+        require(stakedAmount > 0, "StakingContract::migrateStakedTokens - no staked tokens");
+        require(_amount <= stakedAmount, "StakingContract::migrateStakedTokens - amount exceeds staked token balance");
 
-        stakeData.amount = 0;
+        stakeData.amount = stakedAmount.sub(_amount);
 
-        totalStakedTokens = totalStakedTokens.sub(amount);
+        totalStakedTokens = totalStakedTokens.sub(_amount);
 
         require(_newStakingContract.getToken() == token,
             "StakingContract::migrateStakedTokens - staked tokens must be the same");
-        require(token.approve(address(_newStakingContract), amount),
+        require(token.approve(address(_newStakingContract), _amount),
             "StakingContract::migrateStakedTokens - couldn't approve transfer");
 
-        emit MigratedStake(stakeOwner, amount);
+        emit MigratedStake(stakeOwner, _amount);
```

Comment:
This event should include uint256 totalStakedAmount

Reply:
Good point. Will be added after merging #3